### PR TITLE
EOS-13887: Salt pillar doesn't refresh after single node reboot

### PR DESCRIPTION
**Pull request by SomwanshiAnjali**
Signed-off-by: Anjali Somwanshi <anjali.somwanshi@seagate.com>

### CODE REVIEW

**Review by ypise — APPROVED** [review]

**Comment by ypise** [comment]
@andkononykhin 
Would need your attention on this one.

File: srv/components/provisioner/files/glusterfshsaredstorage.service [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[Unit]
+Description=Mount glusterfs sharedstorage
+After=glusterd.service
+Before=salt-minion.service
+Requires=glusterd.service salt-minion.service remote-fs-pre.target local-fs.target
+BindsTo=glusterd.service salt-minion.service
+
+[Service]
+Type=forking
+ExecStart=/usr/libexec/glusterfs/mount-shared-storage.sh
```
Comment (andkononykhin2) [inline srv/components/provisioner/files/glusterfshsaredstorage.service:10]: @SomwanshiAnjali I guess you found some template for that service, are you sure with that command?

File: srv/components/provisioner/config.sls [hunk 2]
```diff
@@ -27,3 +27,16 @@ provisioner_rsyslog_conf_updated:
     - makedirs: True
     - watch_in:
       - service: Start rsyslog
+
+# Always start glusterfshsaredstorage before salt-minion
+Update glusterfshsaredstorage.service:
```
Comment (andkononykhin2) [inline srv/components/provisioner/config.sls:32]: gluster and salt are configured as part of setup stage, can we move that there, e.g. to [client configuration](https://github.com/Seagate/cortx-prvsnr/blob/pre-cortx-1.0/api/python/provisioner/srv/salt/glusterfs/client/config.sls).

**Review by andkononykhin2 — COMMENTED** [review]
@SomwanshiAnjali looks pretty good. I have just concerned regarding code location. Please check the comments. Thank you.

File: srv/components/provisioner/files/glusterfshsaredstorage.service (hunk 1 above)
Comment (SomwanshiAnjali) [reply srv/components/provisioner/files/glusterfshsaredstorage.service:10]: @andkononykhin2 
for this service I referred details from https://github.com/gluster/glusterfs/commits/master/extras/systemd/glusterfssharedstorage.service.in
Also I have checked this script and it just mounts the glusterfs type mounts which are missing on reboot.

File: srv/components/provisioner/files/glusterfshsaredstorage.service [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[Unit]
+Description=Mount glusterfs sharedstorage
+After=glusterd.service
+Before=salt-minion.service
+Requires=glusterd.service salt-minion.service remote-fs-pre.target local-fs.target
```
[the remark this answers is not in the document]
Comment (SomwanshiAnjali) [reply srv/components/provisioner/files/glusterfshsaredstorage.service:5]: yes we can add this as an improvement.

File: srv/components/provisioner/config.sls (hunk 2 above)
Comment (ypise) [reply srv/components/provisioner/config.sls:32]: @andkononykhin 
I am concerned that after update at some point in future if we update gluster services, we would lose this service customization if it is part of `./api/python`.
Being part of `srv/components/provisioner` would take care of update scenario and ensure the customization is applied every time we do an update.

**Review by ypise — APPROVED** [review]

File: srv/components/provisioner/files/glusterfshsaredstorage.service (hunk 1 above)
Comment (ypise) [reply srv/components/provisioner/files/glusterfshsaredstorage.service:10]: @SomwanshiAnjali 
I feel what @andkononykhin2 is trying to say is there's a possibility of this template changing in source which we might miss to capture.
Do evaluate option to insert the lines in existing file, include it if it makes sense; else we continue with template.

File: srv/components/provisioner/files/glusterfshsaredstorage.service (hunk 3 above)
[the remark this answers is not in the document]
Comment (ypise) [reply srv/components/provisioner/files/glusterfshsaredstorage.service:5]: @SomwanshiAnjali 
Please ensure this improvement request is captured.

**Comment by ypise** [comment]
@divya-vijay 
Have a look at this change.

**Pull request merged**